adapter/internal/oasparser/model/mgw_swagger.go

```diff
@@ -264,17 +278,19 @@ func (swagger *MgwSwagger) setXWso2PrdoductionEndpoint() error {
 }
 
 func (swagger *MgwSwagger) setXWso2SandboxEndpoint() error {
-	xWso2APIEndpoints := getXWso2Endpoints(swagger.vendorExtensions, sandboxEndpoints)
+	xWso2APIEndpoints, err := getXWso2Endpoints(swagger.vendorExtensions, sandboxEndpoints)
 	if xWso2APIEndpoints != nil && len(xWso2APIEndpoints) > 0 {
 		swagger.sandboxUrls = xWso2APIEndpoints
-	} else {
+	} else if err != nil {
 		return errors.New("error encountered when extracting endpoints")
```

Comment:
Better if we can provide more details about the endpoint url and etc

Reply:
We already log those specific details. What exact endpoint is logged at the `getHostPortandBasePath` function. When notifying that this API is no longer processed due to an error, we log the API related details at the root level. This is added just to propagate the error.